src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +13,166 @@ class MatchSamplingRate(Transform):
     ) -> None:
         """Initialize the MatchSamplingRate transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature: Reference timeseries feature.
+            feature_columns: key-value pairs of feature columns and
+                the interpolation method to use. The interpolation
+                method can be one of the following:
+                - 'linear'
+                - 'slerp' (to be implemented)
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature = reference_feature
+        self.feature_columns = feature_columns
 
     def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
+        # preserve all constant columns that are not timeseries data
+        non_timeseries_features = [
+            col
+            for col in data.columns
+            if col not in [*self.feature_columns, self.reference_feature]
+        ]
         logger.debug("Matching sampling rate of time series.")
 
-        for i in range(len(data)):
-            reference_timestamps = data[self.reference_timestamps][i]
-            for (
-                feature,
-                timestamp,
-            ) in self.feature_columns_with_timestamps.items():
-                timestamps = data[timestamp][i].to_numpy()
-                feature_data = data[feature][i].to_numpy()
-                resampled_timeseries = interp(
-                    reference_timestamps,
-                    timestamps,
-                    feature_data,
-                ).tolist()
-                data = (
-                    data.lazy()
-                    .with_row_index()
-                    .with_columns(
-                        pl.when(pl.col("index") == i)
-                        .then(resampled_timeseries)
-                        .otherwise(pl.col(feature))
-                        .alias(feature)
-                    )
-                    .drop("index")
-                    .collect()
+        if self.reference_feature not in data.columns:
+            msg = f"Reference {self.reference_feature} not found in the DataFrame."  # noqa: E501
+            raise ValueError(msg)
+
+        features = list(self.feature_columns.keys())
+        reference_df = (
+            data.select(pl.col(self.reference_feature).explode())
+            .unnest(cs.all())
+            .rename(
+                lambda name: self.reference_feature + "_" + name
+                if name != "time"
+                else name
+            )
```

Comment:
We always consider these to be columns named `time` and `value` don't we?

Reply:
yes, this would be an example for the structure that we agreed on: 
```
"feature_b": [
      [
          {
              "time": datetime(
                  2024, 6, 25, 12, 26, 1, 0, tzinfo=UTC
              ),
              "value": {"feature_b_x": 1.2, "feature_b_y": 12.0},
          },
          {
              "time": datetime(
                  2024, 6, 25, 12, 26, 2, 0, tzinfo=UTC
              ),
              "value": {"feature_b_x": 1.4, "feature_b_y": 14.0},
          },
      ]
],
```
We did not agree on limitations regarding the naming of keys inside the content of the `value` key, did we?

Reply:
I meant, you provide a lambda knowing, that you only ever will see a column named `value` and the `time` column you already excluded with the if. So is there actually a need for the lambda function?

Reply:
I changed the input to the rename function, using a dict instead of the lambda function